Zend/zend_string.h

```diff
@@ -405,15 +405,34 @@ static zend_always_inline zend_ulong zend_inline_hash_func(const char *str, size
 	}
 #else
 	/* variant with the hash unrolled eight times */
-	for (; len >= 8; len -= 8) {
-		hash = ((hash << 5) + hash) + *str++;
-		hash = ((hash << 5) + hash) + *str++;
-		hash = ((hash << 5) + hash) + *str++;
-		hash = ((hash << 5) + hash) + *str++;
-		hash = ((hash << 5) + hash) + *str++;
-		hash = ((hash << 5) + hash) + *str++;
-		hash = ((hash << 5) + hash) + *str++;
-		hash = ((hash << 5) + hash) + *str++;
+	for (; len >= 8; len -= 8, str += 8) {
+#if defined(__aarch64__)
+		uint64_t chunk;
+		/* On some architectures it is beneficial to load 8 bytes at a
+		   time and extract each byte with a bit field extract instr. */
+		memcpy(&chunk, str, sizeof(chunk));
```

Comment:
This looks problematic from an endianness perspective -- doesn't aarch64 also support big endian? While technically the specific values does not matter, we do have some places where pregenerated hashes are embedded in the source code that need to be consistent.